### PR TITLE
Fixed IndexOutOfBounds when credential and name are empty (Fixes #88)

### LINKED ISSUE #88: IconManager IndexOutOfBoundsException
On the Galaxy S8 I managed to throw a IndexOutOfBoundsException. After discovering that this app was open source, I started debugging and got the following stacktrace:
```
2018-12-20 22:32:45.281 9124-9124/com.yubico.yubioath E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.yubico.yubioath, PID: 9124
    java.lang.StringIndexOutOfBoundsException: length=0; index=1
        at java.lang.String.substring(String.java:1971)
        at com.yubico.yubioath.ui.main.IconManager.getLetterIcon(IconManager.kt:69)
        at com.yubico.yubioath.ui.main.IconManager.getIcon(IconManager.kt:60)
        at com.yubico.yubioath.ui.main.CredentialAdapter.getView(CredentialAdapter.kt:118)
        at android.widget.AbsListView.obtainView(AbsListView.java:3189)
        at android.widget.ListView.makeAndAddView(ListView.java:2197)
        at android.widget.ListView.fillDown(ListView.java:824)
        at android.widget.ListView.fillFromTop(ListView.java:885)
        at android.widget.ListView.layoutChildren(ListView.java:1924)
        at android.widget.AbsListView.onLayout(AbsListView.java:2961)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:625)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1183)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:870)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:889)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at androidx.appcompat.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:444)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at com.android.internal.policy.DecorView.onLayout(DecorView.java:944)
        at android.view.View.layout(View.java:20836)
        at android.view.ViewGroup.layout(ViewGroup.java:6401)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2948)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2635)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1779)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7810)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
        at android.view.Choreographer.doFrame(Choreographer.java:658)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

### PR DESCRIPTION
**Pull request by BjarnBronsveld**

### CODE REVIEW

**Comment by BjarnBronsveld** [comment]
I've also performed several other improvements like removing experimental coroutines and migrating to Kotlin 1.3.11 and some minor dependency updates.

File: app/src/main/kotlin/com/yubico/yubioath/keystore/ClearingMemProvider.kt [hunk 1]
```diff
@@ -16,16 +12,16 @@ class ClearingMemProvider(private val exec: CoroutineDispatcher) : KeyProvider {
 
     override fun getKeys(deviceId: String): Sequence<StoredSigner> {
         clearJob?.cancel()
-        clearJob = launch(exec) {
-            delay(5, TimeUnit.MINUTES)  //Clear stored passwords after inactivity.
+        clearJob = GlobalScope.launch(exec) {
+            delay(5 * 6)  //Clear stored passwords after inactivity.
```
Comment (BjarnBronsveld) [inline app/src/main/kotlin/com/yubico/yubioath/keystore/ClearingMemProvider.kt]: This line was wrong and needs to be fixed in a newer commit.

**Comment by BjarnBronsveld** [comment]
> Thanks for the contribution!
> 
> I'd like to merge the fix for #88. The other changes however have already been done on a private branch (that will soon be pushed once a few other things are taken care of). Please use separate PRs for unrelated changes.

No problem! Will do in the future. I intended on using seperate PRs, but accidentally screw up. Good to hear! Looking forward to it.

**Comment by dainnilsson** [comment]
Great. I've now merged the first commit.

**Pull request closed without merging**